src/client/java/teammates/client/scripts/sql/DataMigrationForCourseEntitySql.java

```diff
@@ -412,22 +436,97 @@ private teammates.storage.sqlentity.FeedbackResponseComment createFeedbackRespon
+                    oldComment.getLastEditorEmail());
 
         newComment.setCreatedAt(oldComment.getCreatedAt());
         newComment.setUpdatedAt(oldComment.getLastEditedAt());
 
         return newComment;
     }
 
+    // methods for misc migration methods ---------------------------------------------------------------------------------
+
+    private void migrateInstructorEntities(teammates.storage.sqlentity.Course newCourse,
+            Map<String, User> userEmailToUserMap) {
+        List<Instructor> oldInstructors = ofy().load().type(Instructor.class).filter("courseId", newCourse.getId())
+                .list();
+        for (Instructor oldInstructor : oldInstructors) {
+            teammates.storage.sqlentity.Instructor newInstructor = migrateInstructor(newCourse, oldInstructor);
+            userEmailToUserMap.put(oldInstructor.getEmail(), newInstructor);
+        }
+    }
+
+    private teammates.storage.sqlentity.Instructor migrateInstructor(teammates.storage.sqlentity.Course newCourse,
+            Instructor oldInstructor) {
+        InstructorPrivileges newPrivileges;
+        if (oldInstructor.getInstructorPrivilegesAsText() == null) {
+            newPrivileges = new InstructorPrivileges(oldInstructor.getRole());
+        } else {
+            InstructorPrivilegesLegacy privilegesLegacy = JsonUtils
+                    .fromJson(oldInstructor.getInstructorPrivilegesAsText(), InstructorPrivilegesLegacy.class);
+            newPrivileges = new InstructorPrivileges(privilegesLegacy);
+        }
+
+        teammates.storage.sqlentity.Instructor newInstructor = new teammates.storage.sqlentity.Instructor(
+                newCourse,
+                oldInstructor.getName(),
+                oldInstructor.getEmail(),
+                oldInstructor.isDisplayedToStudents(),
+                oldInstructor.getDisplayedName(),
+                InstructorPermissionRole.getEnum(oldInstructor.getRole()),
+                newPrivileges);
+
+        newInstructor.setCreatedAt(oldInstructor.getCreatedAt());
+        newInstructor.setUpdatedAt(oldInstructor.getUpdatedAt());
+        newInstructor.setRegKey(oldInstructor.getRegistrationKey());
+
+        saveEntityDeferred(newInstructor);
+        return newInstructor;
+    }
+
+    private void migrateDeadlineExtensionEntities(teammates.storage.sqlentity.Course newCourse,
+            Map<String, teammates.storage.sqlentity.FeedbackSession> feedbackSessionNameToFeedbackSessionMap,
+            Map<String, User> userEmailToUserMap) {
+        List<DeadlineExtension> oldDeadlineExtensions = ofy().load().type(DeadlineExtension.class)
+                .filter("courseId", newCourse.getId())
+                .list();
+
+        for (DeadlineExtension oldDeadlineExtension : oldDeadlineExtensions) {
+            User newUser = userEmailToUserMap.get(oldDeadlineExtension.getUserEmail());
+            if (newUser == null) {
+                continue;
+            }
```

Comment:
If we can't get the user, is this a problem / throw an exception?

Reply:
Actually I was thinking of logging it but I am waiting for a logging function from my teammates